convert_checkpoints.py

```diff
@@ -268,6 +275,84 @@ def _load_from_local(input_ckpt_dir: epath.Path):
+    )
+  checkpoint = {}
+  for st_f in safetensors_files:
+    with safe_open(st_f, framework="pt", device="cpu") as f:
+      for key in f.keys():
+        if "inv_freq" in key:
+          # Don't include 'rotary_emb.inv_freq' in the converted
+          # checkpoint, because in JetStream implementation we
+          # precompute it during weight loading.
+          continue
+        new_key = key
+        # Remove 'model.' prefix for all weights.
+        prefix_to_remove = "model."
+        if key.startswith(prefix_to_remove):
+          new_key = new_key.removeprefix(prefix_to_remove)
+
+        # Weight name substring mapping between hf and jetstream.
+        _load_hf_llama_weight.hf_to_jetstream_keys_mapping = {
+            "lm_head": "output",
+            "embed_tokens": "tok_embeddings",
+            "input_layernorm": "attention_norm",
+            "post_attention_layernorm": "ffn_norm",
+            "self_attn.q_proj": "attention.wq",
+            "self_attn.k_proj": "attention.wk",
+            "self_attn.v_proj": "attention.wv",
+            "self_attn.o_proj": "attention.wo",
+            "mlp.gate_proj": "feed_forward.w1",
+            "mlp.down_proj": "feed_forward.w2",
+            "mlp.up_proj": "feed_forward.w3",
+            "model.norm.weight": "norm.weight",
+        }
+        found_substute = False
+        for (
+            hf_weight_key
+        ) in _load_hf_llama_weight.hf_to_jetstream_keys_mapping.keys():
+          if hf_weight_key in key:
+            jet_stream_key = _load_hf_llama_weight.hf_to_jetstream_keys_mapping[
+                hf_weight_key
+            ]
+            new_key = new_key.replace(hf_weight_key, jet_stream_key)
+            found_substute = True
+            break
+        assert found_substute, f"No substitute name found for {key}."
+        print(f"convert weight name {key} to {new_key}.")
+        weight_tensor = f.get_tensor(key)
+        if weight_tensor.dtype == torch.float16:
+          # JetStream expects bf16 weight, since activation is in bf16
+          # float16 x bf16 will hit mix precision assertion.
+          weight_tensor = weight_tensor.to(torch.bfloat16)
+          print(f"convert weight name {new_key} from float16 to bfloat16.")
+        if "wq" in new_key or "wk" in new_key:
+          # In HF weight, wq and wk are interleaved differently
+          weight_shape = weight_tensor.shape
+          weight_tensor = (
+              weight_tensor.reshape(-1, 2, 64, weight_shape[1])
+              .transpose(1, 2)
+              .reshape(weight_shape)
+          )
+        checkpoint[new_key] = weight_tensor
+  return [checkpoint], None
+
+
+def _load_from_local(input_ckpt_dir: epath.Path):
+  if not _FROM_HF.value:
+    return _load_orig_llama_weight(input_ckpt_dir)
+  else:
+    assert (
+        not FLAGS.quantize_weights
+    ), "Quantization not supported for HF checkpoint."
+    return _load_hf_llama_weight(input_ckpt_dir)
```

Comment:
Did you test the llama2-70B model?

Reply:
No I didn't, since I haven't set up multi host yet. But I will do that later